### PR TITLE
Add Error Prone static analysis tool

### PR DESCRIPTION
**Pull request by BohdanDudkaTD**
This PR adds `Error Prone` as a static analysis tool for Java.
Add its dependencies to the `build.gradle` in the `server` module.

Also, fixed issue with incorrect mapping of _out/_ folder in the .gitignore file.
All variables with dependencies versions were moved to `gradle.properties` file in the server module.

### CODE REVIEW

**Comment by BohdanDudkaTD** [comment]
@AndriiSameliuk, PTAL.

**Comment by BohdanDudkaTD** [comment]
@Oleh-Stefanovych, @okucheruk, PTAL.

File: server/build.gradle [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
     id 'idea'
+    id "net.ltgt.errorprone" version "0.8.1"
 }
 
 group 'io.javaclasses'
 version '1.0-SNAPSHOT'
 
 ext.slf4j = '1.7.28'
 ext.jUnitVersion = '5.5.2'
+ext.errorproneCoreVersion = '2.3.3'
```
Comment (okucheruk) [inline server/build.gradle]: We can simplify this to 
```
ext {
    slf4j = '1.7.28'
    jUnitVersion = '5.5.2'
    errorproneCoreVersion = '2.3.3'
}
```

**Comment by BohdanDudkaTD** [comment]
@okucheruk, PTAL.
Fixed issues:
- create `gradle.properties`  file and extract all dependencies versions there;
- rename `slf4j` so it's more consistent.

**Comment by BohdanDudkaTD** [comment]
@Oleg-Melnik, @armiol, PTAL.

**Pull request merged**